### PR TITLE
Processor unit tests

### PR DESCRIPTION
**Pull request by guozhangwang**
@ymatsuda 

"./gradlew streaming:jar" succeeds again, finally.

1) Fix the StreamTaskTest.
2) Fix ProcessorJob example code.

### CODE REVIEW

File: stream/src/main/java/org/apache/kafka/streaming/kstream/internals/KStreamJoin.java [hunk 1]
```diff
@@ -122,6 +129,34 @@ public void process(K key, V1 value) {
             }
         }
 
+        private boolean joinable() {
+            Set<TopicPartition> partitions = ((ProcessorContextImpl) this.context).task().partitions();
```
Comment (ymatsuda) [inline stream/src/main/java/org/apache/kafka/streaming/kstream/internals/KStreamJoin.java]: Didn't this break a KStream test? KStream tests use MockProcessorContext. 
Why did you move joinable() from ProcessorContext to KStreamJoin?

Comment (guozhangwang) [inline stream/src/main/java/org/apache/kafka/streaming/kstream/internals/KStreamJoin.java]: I feel this function is specific to KStream not in Processor, as its checking logic is sort-of specific to KStream's definitions. Which tests need MockProcessorContext.joinable() calls?

Comment (ymatsuda) [inline stream/src/main/java/org/apache/kafka/streaming/kstream/internals/KStreamJoin.java]: KStreamJoin should not care how the joinability is computed. So, having joinable() in ProcessorContext is actually good.

Anything that makes tests harder is bad since it is an indication of a bad code organization.

Comment (ymatsuda) [inline stream/src/main/java/org/apache/kafka/streaming/kstream/internals/KStreamJoin.java]: Join is not a KStream specific concept.

Comment (guozhangwang) [inline stream/src/main/java/org/apache/kafka/streaming/kstream/internals/KStreamJoin.java]: OK, will change it then.

File: stream/src/main/java/org/apache/kafka/streaming/kstream/internals/KStreamJoin.java [hunk 2]
```diff
@@ -17,13 +17,20 @@
...
 import org.apache.kafka.streaming.kstream.Window;
 import org.apache.kafka.streaming.processor.ProcessorDef;
+import org.apache.kafka.streaming.processor.internals.ProcessorContextImpl;
 
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
```
Comment (ymatsuda) [inline stream/src/main/java/org/apache/kafka/streaming/kstream/internals/KStreamJoin.java]: clean up imports

Comment (guozhangwang) [inline stream/src/main/java/org/apache/kafka/streaming/kstream/internals/KStreamJoin.java]: There is no unused imports, I will group o.a.k.s.kstream and o.a.k.s.processor.

Comment (ymatsuda) [inline stream/src/main/java/org/apache/kafka/streaming/kstream/internals/KStreamJoin.java]: Sorry, I meant the clean up after you move joinable() back to ProcessContext.

**Comment by guozhangwang** [comment]
@ymatsuda 

Now "./gradlew stream:test" works, but 10 out of 29 tests fails.. Could we work in parallel to fix them after checking this in?

**Comment by ymatsuda** [comment]
OK. Let's merge your PR.

**Comment by guozhangwang** [comment]
@ymatsuda do you have other comments for this one?
